eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/list/ThrowingMutableListFactory.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2018 Goldman Sachs.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.api.factory.list;
+
+import org.eclipse.collections.api.block.function.Function0;
+import org.eclipse.collections.api.list.MutableList;
+
+public class ThrowingMutableListFactory implements MutableListFactory
+{
+    private final String error;
+
+    public ThrowingMutableListFactory(String error)
+    {
+        this.error = error;
+    }
+
+    @Override
+    public <T> MutableList<T> empty()
+    {
+        throw new IllegalStateException(this.error);
```

Comment:
The Throwing*Factories implement the same interfaces, like MutableListFactory in this case. I pass in the error message in the constructor, since I use a different error message for 0 implementations found, and for >2 implementations found. All methods throw IllegalStateException, but I'm not sure if that's the right choice.